lib/matplotlib/colors.py

```diff
@@ -1368,8 +1368,14 @@ def inverse(self, value):
     def autoscale(self, A):
         """Set *vmin*, *vmax* to min, max of *A*."""
         A = np.asanyarray(A)
-        self.vmin = A.min()
-        self.vmax = A.max()
+        # Ensure that A has non-zero size to compute min/max.
+        # Otherwise put in filler values.
+        if A.size:
+            self.vmin = A.min()
+            self.vmax = A.max()
+        else:
+            self.vmin = 0.0
+            self.vmax = 0.0
```

Comment:
Setting these values to zero seems almost certainly wrong; the pre-#21989 behavior was to set them to np.ma.masked (I think?) which also seems a bit strange but already better.